api/envoy/extensions/filters/http/stateful_session/v3/stateful_session.proto

```diff
@@ -0,0 +1,56 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.stateful_session.v3;
+
+import "envoy/config/core/v3/extension.proto";
+import "envoy/config/core/v3/health_check.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.stateful_session.v3";
+option java_outer_classname = "StatefulSessionProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/stateful_session/v3;stateful_sessionv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Stateful Session Filter]
+// This filter sets an override host based on the extensible session state and updates the
+// session state based on the final selected upstream host.
+// [#extension: envoy.filters.http.stateful_session]
+
+// [#comment:TODO(wbpcode): To support global feature enabled or disabled flag.]
+message StatefulSession {
+  // The expected health status of the host. Suppose a session is assigned to the host X.
```

Comment:
I don't think any user browsing the docs will know how to make use of this. You're going to have to write some higher level docs explaining when you want stateful session, how you might go about using health and so on. I'm pretty confused why we need to be micromanaging health here in an HTTP filter when this is something that should be done further upstream.

Reply:
This state is only an expectation. It is eventually passed to the `LoadBalancer` through the `overrideHost` interface of the `LocalBalancerCallback` and affects the final Load balancing result.

And why we implemented it with HTTP filter:

> I wonder if we'd be better off putting most of the stateful session implementation in an http filter? Certainly the cookie lookup and generation could be done there. We'd need a way for the session filter to pass an OverrideHost (or equivalent) to the LoadBalancerContext implementation in the router filter. We already have FilterState and dynamic metadata available in StreamInfo or we could introduce a method in Stream[Encoder]FilterCallbacks.

> The advantage to using a separate filter would be to reuse the existing implementation of the route's typed_per_filter_config to allow stateful sessions to be enabled/disabled per route (and perhaps even to allow more advanced config difference across routes, like altering the allowed health or even the cookie name).

> +1 from me if we can easily implement this as a filter + FS, I think that would make it easier to use this feature in various creative ways vs it being specifically for stateful load balancing

Previous comments from other reviewers.

Reply:
> I don't think any user browsing the docs will know how to make use of this.

I will update the docs make it more clear.

Reply:
@htuch Hi, I've added a document explaining what this filter does and how to use it. Can you see what else needs to be optimized? 😄

Reply:
Thanks, these docs are great. Can you link to the docs from here?

Reply:
get it.